components/job-orchestration/job_orchestration/scheduler/utils.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+from contextlib import closing
+from typing import List, Optional
+
+from clp_py_utils.clp_config import (
+    COMPRESSION_JOBS_TABLE_NAME,
+    COMPRESSION_TASKS_TABLE_NAME,
+    QUERY_JOBS_TABLE_NAME,
+    QUERY_TASKS_TABLE_NAME,
+)
+from clp_py_utils.sql_adapter import SQL_Adapter
+from job_orchestration.scheduler.constants import (
+    CompressionJobStatus,
+    CompressionTaskStatus,
+    QueryJobStatus,
+    QueryTaskStatus,
+    SchedulerType,
+)
+
+
+def kill_hanging_jobs(sql_adapter: SQL_Adapter, scheduler_type: str) -> Optional[List[str]]:
+    if SchedulerType.COMPRESSION == scheduler_type:
+        jobs_table_name = COMPRESSION_JOBS_TABLE_NAME
+        job_status_running = CompressionJobStatus.RUNNING
+        job_status_failed = CompressionJobStatus.FAILED
+        tasks_table_name = COMPRESSION_TASKS_TABLE_NAME
+        task_status_running = CompressionTaskStatus.RUNNING
+        task_status_failed = CompressionTaskStatus.FAILED
+    elif SchedulerType.QUERY == scheduler_type:
+        jobs_table_name = QUERY_JOBS_TABLE_NAME
+        job_status_running = QueryJobStatus.RUNNING
+        job_status_failed = QueryJobStatus.FAILED
+        tasks_table_name = QUERY_TASKS_TABLE_NAME
+        task_status_running = QueryTaskStatus.RUNNING
+        task_status_failed = QueryTaskStatus.FAILED
```

Comment:
This looks pretty ugly. I have discussed with Zhihao and we both agree that having a class managing the Tablename, job/task status might be the right way to go but it would require more refactoring.

For now, I still feel this is better than asking user to passin all 6 arguments by themselves.